File: backend/app/core/permissions.py
from app.models.enums import (
    AnalyticsScope,
    BalancesScope,
    GroupRole,
    MembersScope,
    OrdersScope,
    PermissionType,
    RestaurantsScope,
)

# Role preset definitions mapping role -> {permission_type: level}
GROUP_ROLE_PRESETS: dict[GroupRole, dict[PermissionType, str]] = {
    GroupRole.ADMIN: {
        PermissionType.MEMBERS: MembersScope.EDITOR,
        PermissionType.ORDERS: OrdersScope.EDITOR,
        PermissionType.BALANCES: BalancesScope.EDITOR,
        PermissionType.ANALYTICS: AnalyticsScope.VIEWER,
        PermissionType.RESTAURANTS: RestaurantsScope.EDITOR,
    },
    GroupRole.SUPERVISOR_MEMBER: {
        PermissionType.MEMBERS: MembersScope.VIEWER,
        PermissionType.ORDERS: OrdersScope.INITIATOR,
        PermissionType.BALANCES: BalancesScope.VIEWER,
        PermissionType.ANALYTICS: AnalyticsScope.VIEWER,
        PermissionType.RESTAURANTS: RestaurantsScope.VIEWER,
    },
    GroupRole.MEMBER: {
        PermissionType.MEMBERS: MembersScope.NONE,
        PermissionType.ORDERS: OrdersScope.PARTICIPANT,
        PermissionType.BALANCES: BalancesScope.NONE,
        PermissionType.ANALYTICS: AnalyticsScope.NONE,
        PermissionType.RESTAURANTS: RestaurantsScope.VIEWER,
    },
}

File: frontend/src/constants/permissions.ts
export const PERMISSION_TYPE = {
  MEMBERS: "members",
  ORDERS: "orders",
  BALANCES: "balances",
  ANALYTICS: "analytics",
  RESTAURANTS: "restaurants",
} as const;

export const MEMBERS_SCOPE = {
  EDITOR: "editor",
  VIEWER: "viewer",
  NONE: "none",
} as const;

export const ORDERS_SCOPE = {
  EDITOR: "editor",
  INITIATOR: "initiator",
  PARTICIPANT: "participant",
} as const;

export const BALANCES_SCOPE = {
  EDITOR: "editor",
  VIEWER: "viewer",
  NONE: "none",
} as const;

export const ANALYTICS_SCOPE = {
  VIEWER: "viewer",
  NONE: "none",
} as const;

export const RESTAURANTS_SCOPE = {
  EDITOR: "editor",
  VIEWER: "viewer",
} as const;
